mlflow/gateway/handlers.py

```diff
@@ -0,0 +1,108 @@
+from enum import Enum
+from pathlib import Path
+from pydantic import BaseModel, Field, validator, parse_obj_as
+from pydantic.json import pydantic_encoder
+from typing import Optional, Union, List
+import yaml
+import json
+
+
+class Provider(str, Enum):
+    UNSPECIFIED_PROVIDER = "unspecified"
+    OPENAI = "openai"
+    ANTHROPIC = "anthropic"
+    BARD = "bard"
+
+
+class RouteType(str, Enum):
+    UNSPECIFIED = "unspecified"
+    LLM_V1_INSTRUCT = "llm/v1/instruct"
+    LLM_V1_CHAT = "llm/v1/chat"
+
+
+class OpenAIConfig(BaseModel):
+    openai_api_key: str = Field(..., alias="openai_api_key")
```

Comment:
Should we support reading API keys from environment variables?

Reply:
pending commit has this support